src/Kentico.Xperience.Lucene/Services/Implementations/DefaultLuceneIndexService.cs

```diff
@@ -10,8 +10,7 @@ namespace Kentico.Xperience.Lucene.Services.Implementations;
 public class DefaultLuceneIndexService : ILuceneIndexService
 {
     private const LuceneVersion LUCENE_VERSION = LuceneVersion.LUCENE_48;
-
-    public TResult UseWriter<TResult>(LuceneIndex index, Func<IndexWriter, TResult> useIndexWriter)
+    public TResult UseWriter<TResult>(LuceneIndex index, Func<IndexWriter, TResult> useIndexWriter, OpenMode openMode = OpenMode.CREATE_OR_APPEND)
```

Comment:
Is `openMode` supposed to be used in this method?

Reply:
It is optional, it should be CREATE_OR_APPEND in most cases with exception of index rebuild. My original idea was to recreate the index from scratch bay rewriting it, but if I just open the action it is not rewritten. That I why I use DeleteAll().

Reply:
What I mean is, the parameter `openMode` for this method is not used _in the method_. So, I'm wondering if that was intentional and if it was, why is it in the method signature?